include/deal.II/lac/la_vector.h

```diff
@@ -0,0 +1,440 @@
+     * only allowed if @p s is equal to zero.
+     */
+    Vector<Number> &operator= (const Number s);
+
+    /**
+     * Multiply the entire vector by a fixed factor.
+     */
+    virtual VectorSpaceVector<Number> &operator*= (const Number factor);
+
+    /**
+     * Divide the entire vector by a fixed factor.
+     */
+    virtual VectorSpaceVector<Number> &operator/= (const Number factor);
+
+    /**
+     * Add the vector @p V to the present one.
+     */
+    virtual VectorSpaceVector<Number> &operator+= (const VectorSpaceVector<Number> &V);
+
+    /**
+     * Substract the vector @p V from the present one.
+     */
+    virtual VectorSpaceVector<Number> &operator-= (const VectorSpaceVector<Number> &V);
+
+    /**
+     * Return the scalar product of two vectors.
+     */
+    virtual Number operator* (const VectorSpaceVector<Number> &V);
+
+    /**
+     * Add @p a to all components. Note that @p a is a scalar not a vector.
+     */
+    virtual void add(const Number a);
+
+    /**
+     * Simple addition of a multiple of a vector, i.e. <tt>*this += a*V</tt>.
+     */
+    virtual void add(const Number a, const VectorSpaceVector<Number> &V);
+
+    /**
+     * Multiple addition of a multiple of a vector, i.e. <tt>*this += a*V+b*W</tt>.
+     */
+    virtual void add(const Number a, const VectorSpaceVector<Number> &V,
+                     const Number b, const VectorSpaceVector<Number> &W);
+
+    /**
+     * Scaling and simple vector addition, i.e. <tt>*this = s*(*this)+V</tt>.
+     */
+    virtual void sadd(const Number s, const VectorSpaceVector<Number> &V);
+
+    /**
+     * Scaling and simple addition of a multiple of a vector, i.e. <tt>*this =
+     * s*(*this)+a*V</tt>.
+     */
+    virtual void sadd(const Number s, const Number a,
+                      const VectorSpaceVector<Number> &V);
+
+    /**
+     * Scale each element of this vector by the corresponding element in the
+     * argument. This function is mostly meant to simulate multiplication (and
+     * immediate re-assignement) by a diagonal scaling matrix.
+     */
+    virtual void scale(const VectorSpaceVector<Number> &scaling_factors);
+
+    /**
+     * Assignement <tt>*this = a*V</tt>.
+     */
+    virtual void equ(const Number a, const VectorSpaceVector<Number> &V);
+
+    /**
+     * Return the l<sub>1</sub> norm of the vector (i.e., the sum of the
+     * absolute values of all entries).
+     */
+    virtual typename VectorSpaceVector<Number>::real_type l1_norm();
+
+    /**
+     * Return the l<sub>2</sub> norm of the vector (i.e., the square root of
+     * the sum of the square of all entries among all processors).
+     */
+    virtual typename VectorSpaceVector<Number>::real_type l2_norm();
+
+    /**
+     * Return the maximum norm of the vector (i.e., the maximum absolute
+     * value among all entries and among all processors).
+     */
+    virtual typename VectorSpaceVector<Number>::real_type linfty_norm();
+
```

Comment:
I think that we decided when we created the ```VectorSpaceVector``` class that we will first try to have the smallest interface and add thing like ```mean_value``` only if it is necessary.